iothub_client/src/iot_smarthome_client.c

```diff
@@ -562,6 +568,35 @@ static void OnRecvCallbackForMethodResp(const IOT_SH_CLIENT_HANDLE handle, const
                 LogError("Unexpected response. %d", status);
             }
         }
+        else if (strcmp(methodName, METHOD_GET_SUB_DEVICES) == 0)
+        {
+            if (status == 200)
+            {
+                SHADOW_SUB_DEVICES *subDevices = malloc(sizeof(SHADOW_SUB_DEVICES *));
+                size_t count = json_array_get_count(payloadArray);
+                subDevices->count = count;
+                subDevices->puids = malloc(sizeof(char *) * count);

```

Comment:
please add check whether memory allocated success? it will fail on some resource constraint device